### ISSUE TITLE
HTTP: Make session available in HttpRequestBuilder.transformResponse

### ISSUE THREAD

**Issue by robinvandenberg** [gatling/gatling#3496]
This has been requested before : 
https://github.com/gatling/gatling/issues/3461
https://stackoverflow.com/questions/50748527/accessing-session-variable-inside-a-response-transformer-in-gatling

My usecase is as follows:
The response is encrypted. The response can be decrypted with a key that is negotiated while doing API-calls. Thus, I have a decryptionkey - per user - in the session.

I'd like to be able to do .transformResponse() and provide a function that accepts the session so I can decrypt the response. Then, the decrypted response will be propagated to the check pipeline so I can do multiple 'checks' with 'jsonPath' for example.
I currently have a workaround but that requires multilpe (ugly) statements:
check(ByteArrayBody.transform(.. put decrypted body in session .. ))
check(ByteArrayBody.transform(.. get decrypted body from session and do an assertion on it))
Unfortunately, this workaround is not really fluent, because you would like to use things like 'jsonPath' for example.

**Comment by slandelle** [comment]
`transformResponse` was intended for unpacking, eg protobuf.

Supporting this use case would be non trivial, and I'm reluctant to implement a feature if it serves a purpose for one single user.
Se before deciding if this should be implemented, I'd like to have it requested for more people.

**Comment by javaigua** [comment]
Please do implement. 

From my perspective this would be complementary to `signatureCalculator` for the same scenario that @robinvandenberg mentions above.

**Comment by slandelle** [comment]
@javaigua Could you please explain your use case?

**Comment by javaigua** [comment]
Thank you for your quick response @slandelle 

The RESTful API that I'm testing handles encryption on production. First I developed all my test against a sandbox with this feature disabled. Now that I want to handle encryption and hit production I believe gatling's DSL is falling short to provide developers a hook to fully transform the response.

In my code I was able to encapsulate per-user encryption in just one `signatureCalculator` call, but then if I'm not able to access the user session in `transformResponse` there is no clean and fluent way to decrypt the contextual response.
